### ISSUE TITLE
git worktree support

### ISSUE THREAD

**Issue by Nezteb** [git-town/git-town#2134]
Currently if you're using `git worktree`, some `git-town` commands will fail.

Example:
- Have a repo with a `develop` branch.
- Create new directory + worktree for a new `feature-abc` branch and open it.
- Run `git-town set-parent-branch` and set it to `develop`.
- Add some commits to the `feature-abc` branch.
- Run `git-town sync` and get this error:
```
❯ git sync

[feature-abc] git fetch --prune --tags

[feature-abc] git checkout develop
fatal: 'develop' is already checked out at './path/to/repo'

Error: cannot check out branch "develop" in repo ".": exit status 128

To abort, run "git-town abort".
To continue after having resolved conflicts, run "git-town continue".
To continue by skipping the current branch, run "git-town skip".
```

Given that git worktrees are in completely different directories, when `git-town` tries to sync the parent branch with its upstream by first checking out the parent branch, this fails.

The only option I can think of right now is to have `git-town sync` have a configuration option to not sync the local parent branch and instead run something like `git <rebase|merge> origin/<parent-branch>`, which in my case would be `git merge origin/develop`. Obviously this has the downside that your `develop` worktree remains out-of-date, but I can't think of a better option.

**Comment by Nezteb** [comment]
Another quirk is that the list of branches changes slightly when using worktrees, so the branch picker in `git-town` gets confused:
<img width="946" alt="Screenshot 2023-03-06 at 8 14 45 PM" src="https://user-images.githubusercontent.com/3588798/223319432-1e0ea139-3967-4ca2-8720-41de4caa922e.png">

The branch is called `develop` but a worktree shows a `+` sign next to any branch that is set up as another worktree, and `git-town` will happily set the value for that branch's parent as `+ develop`, causing errors. 😅

**Comment by kevgo** [comment]
Thanks for reporting this. You are right, Git worktree support is a missing feature right now. Too bad that worktrees don't allow checking out the same branch in different worktrees. I assume Git has good reasons to enforce this. 

Support for worktrees would be relatively easy to add:

(1): Run `git worktree list` on startup to get an overview of the active worktrees and which branches they have checked out.

(2a): Omit checking out these branches when doing Git Town stuff, and sync with `origin/<parent-branch>` instead (as you propose)

An alternative to (2a) would be:

(2b): sync branches that are checked out in another worktree the hard way:
  - cd into the directory containing the worktree with the branch we want to sync
  - stash away the open changes there
  - sync the branch there
  - un-stash the open changes there
  - cd back

Option (2b) sounds complex and likely has many edge cases that can cause issues. So option (2a) might be the way to go here.

**Comment by kevgo** [comment]
FYI this is now available in [v11.1](https://github.com/git-town/git-town/releases/tag/v11.1.0). Enjoy!

**Comment by vectro** [comment]
As a second step, it would be great if git-town supported something analogous to Jane Street's [Iron Workspaces](https://blog.janestreet.com/putting-the-i-back-in-ide-towards-a-github-explorer/) (scroll to "The lifecycle of a pull request in Feature Explorer").
